nrf_modem: v2.0.0 (#734)
nrfconnect/sdk-nrfxlib: nrf_modem/doc/ug_nrf_modem_porting_os.rst

This function is called by the Modem library when a timed context or sleep is required.
A blind return value of 0 will make all the Modem library operations always blocking.

.. note::

>    To ensure all waiting threads are awake on :c:func:`nrf_modem_event_notify` function, the waiting threads must not be cleared on :c:func:`nrf_modem_os_init` or :c:func:`nrf_modem_os_shutdown` functions.

Reviewer: ```suggestion
   To ensure all waiting threads are awake on the  :c:func:`nrf_modem_event_notify` function, the waiting threads must not be cleared on :c:func:`nrf_modem_os_init` or :c:func:`nrf_modem_os_shutdown` functions.
```

*Required actions* to be taken that do not block the operations:

* Start counting the time (this can be based on a Timer or Thread for instance).
* In the case of a call to :c:func:`nrf_modem_os_event_notify` function, all waiting threads must wake up, reporting back the remaining time of the timer through the :c:var:`timeout` variable.

> The following points decides the *Function return value*:

Reviewer: ```suggestion
The following points decide the *Function return value*:
```

> * If the modem is not initialized, that is if :c:func:`nrf_modem_is_initialized` return false, return ``-NRF_ESHUTDOWN``.

Reviewer: ```suggestion
* If the modem is not initialized, that is, if :c:func:`nrf_modem_is_initialized` returns false, function return value will be ``-NRF_ESHUTDOWN``.
```

> * If there is a time out, return ``-NRF_EAGAIN``.
> * Else return ``0``.

Reviewer: ```suggestion
* If there is a time out, function return value will be ``-NRF_EAGAIN``.
* In all other cases, function return value will be ``0``.
```

This function is called by the Modem library when a timed context or sleep is required.
A blind return value of 0 will make all the Modem library operations always blocking.

.. note::
   The calls to :c:func:`nrf_modem_os_timedwait` function must be able to enter a waiting state regardless of the modem library and modem OS initialization status.

>    Hence, it must be possible to call the :c:func:`nrf_modem_os_timedwait` function before any call to the :c:func:`nrf_modem_os_init` function.

Author: This is no longer the case, right @eivindj-nordic?

Reviewer 2: true

Author: thanks for clarifying

Author: Removed text

   To ensure all waiting threads are awake on :c:func:`nrf_modem_event_notify` function, the waiting threads must not be cleared on :c:func:`nrf_modem_os_init` or :c:func:`nrf_modem_os_shutdown` functions.

*Required actions* to be taken that do not block the operations:

* Start counting the time (this can be based on a Timer or Thread for instance).
* In the case of a call to :c:func:`nrf_modem_os_event_notify` function, all waiting threads must wake up, reporting back the remaining time of the timer through the :c:var:`timeout` variable.

The following points decides the *Function return value*:

* If the modem is not initialized, that is if :c:func:`nrf_modem_is_initialized` return false, return ``-NRF_ESHUTDOWN``.

> * If there is a time out, return ``-NRF_ETIMEDOUT``.

Author: This is now `-NRF_EAGAIN` right @eivindj-nordic ?

Reviewer 2: Yes, `-NRF_ETIMEDOUT` is replaced with `-NRF_EAGAIN`

* Else return ``0``.

nrf_modem_os_event_notify
=========================

> This function is called by the Modem library when an event occurs and all threads waiting for :c:func:`nrf_modem_os_timedwait` function wakes up.

Author: This is not correct anymore. The threads do not wait _for_ `nrf_modem_os_timedwait()`, they wait _in_ `nrf_modem_os_timedwait()`, as the original text said. Also, the threads _wake_ up.
